java/src/main/java/com/google/appengine/tools/mapreduce/impl/shardedjob/ShardedJobRunner.java

```diff
@@ -333,27 +327,34 @@ private void handleLockHeld(String taskId, ShardedJobStateImpl<T> jobState,
             "Resuming after abandon lock for " + taskId + " on slice: "
                 + taskState.getSequenceNumber()), true);
       } else {
-        retryState = handleShardFailure(jobState, taskState, new RuntimeException(
-          "Lock for " + taskId + " expired on slice: " + taskState.getSequenceNumber()));
+        retryState = handleSliceFailure(jobState, taskState, new RuntimeException(
+          "Resuming after abandon lock for " + taskId + " on slice: "
+            + taskState.getSequenceNumber()), true);
+//        retryState = handleShardFailure(jobState, taskState, new RuntimeException(
+//          "Lock for " + taskId + " expired on slice: " + taskState.getSequenceNumber()));
       }
       updateTask(jobState, taskState, retryState, false);
     }
   }
 
+  /**
+   * determines whether a given GAE request was completed by querying against Logs Service
+   * @param requestId
+   * @return whether request is known to have been completed
+   */
   private static boolean wasRequestCompleted(String requestId) {
     if (requestId != null) {
-      LogQuery query = LogQuery.Builder.withRequestIds(Collections.singletonList(requestId));
-      try {
-        for (RequestLogs requestLog : LOG_SERVICE.fetch(query)) {
-          if (requestLog.isFinished()) {
-            log.info("Previous un-released lock for request " + requestId + " has finished");
-            return true;
-          }
-        }
-      } catch (LogServiceException ex) {
-        // consider any log-fetch failure as if request is not known to be completed
-        log.log(Level.FINE, "Failed to query log service for request " + requestId, ex);
-      }
+      //previously, this checked against GAE LogService; this seems to no longer work as-expected
```

Comment:
I thought about doing this too, but I don't get why it stopped working. No libraries changed... I'll try with your branch and mine together and see

Reply:
i think it could be related to credential changes; GAE libraries all just magically pick up credentials. Message you had in logs (no project id) suggests that's not working as expected, at least in dev. so who knows - GAE change, just a dev thing, not sure ...